eva-accession-pipeline/src/test/java/uk/ac/ebi/eva/accession/pipeline/io/AccessionReportWriterTest.java

```diff
@@ -240,4 +276,56 @@ private void assertHeaderIsNotWrittenTwice(File output) throws IOException {
                        .map(variant -> new AccessionWrapper<ISubmittedVariant, String, Long>(ACCESSION, HASH, variant))
                        .collect(Collectors.toList());
     }
+
+    @Test
+    public void writeChromosomeReplacedFromContig() throws IOException {
+        assertContigReplacement(CONTIG_1, ContigNaming.SEQUENCE_NAME, CHROMOSOME_1);
+    }
+
+    private void assertContigReplacement(String originalContig, ContigNaming requestedReplacement,
+                                         String replacementContig) throws IOException {
+        SubmittedVariant variant = new SubmittedVariant("accession", TAXONOMY, "project", originalContig,
+                                                        START_1, "", ALTERNATE, CLUSTERED_VARIANT,
+                                                        SUPPORTED_BY_EVIDENCE, MATCHES_ASSEMBLY, ALLELES_MATCH,
+                                                        VALIDATED, null);
+
+        AccessionWrapper<ISubmittedVariant, String, Long> accessionWrapper = new AccessionWrapper<>(ACCESSION, "hash-1",
+                                                                                                    variant);
+
+        AccessionWrapperComparator accessionWrapperComparator = new AccessionWrapperComparator(
+                Collections.singletonList(variant));
+
+        accessionReportWriter = new AccessionReportWriter(output, fastaSequenceReader, contigMapping,
+                                                          requestedReplacement);
+        accessionReportWriter.open(new ExecutionContext());
+        accessionReportWriter.write(Collections.singletonList(accessionWrapper), accessionWrapperComparator);
+
+        assertEquals(replacementContig, getFirstVariantLine(output).split("\t")[CHROMOSOME_COLUMN_VCF]);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void writeContigWithoutEquivalent() throws IOException {
+        String contigMissingInAssemblyReport = "contig_missing_in_assembly_report";
+        assertContigReplacement(contigMissingInAssemblyReport, ContigNaming.SEQUENCE_NAME,
+                                contigMissingInAssemblyReport);
+    }
+
+    @Test
+    public void writeContigIgnoringNonAssembledMolecule() throws IOException {
```

Comment:
I think it would be a bit more clear this way:
```suggestion
    public void writeContigWithNoAvailableAssembledMolecule() throws IOException {
```

Reply:
"assembled-molecule" and "unlocalized-scaffold" are the flags, "assigned molecule" is the contig synonym. I'll use writeContigWithNoAvailableAssignedMolecule